src/nextjournal/clerk/analyzer.clj

```diff
@@ -559,22 +602,24 @@
                            (vary-meta dissoc :type)))
                  form))
 
-(defn hash-codeblock [->hash {:as codeblock :keys [hash form id deps vars]}]
-  (when (and (seq deps) (not (ifn? ->hash)))
-    (throw (ex-info "`->hash` must be `ifn?`" {:->hash ->hash :codeblock codeblock})))
-  (let [hashed-deps (into #{} (map ->hash) deps)]
```

Comment:
Hash is now computed using from graph nodes, rather than the block deps. I removed the assertion as it seemed confusing to me. The check should rather be that all deps has been hashed at this point.

Reply:
> The check should rather be that all deps has been hashed at this point

If I introduce the same warning on `main`, by analyzing for instance `n.c.viewer` we get: 
```
Missing hash for dep clojure.core/class of form (defrecord ViewerEval [form])
Missing hash for dep clojure.core/declare of form (declare present+reset!)
Missing hash for dep clojure.core/class of form (defrecord ViewerEval [form])
Missing hash for dep nextjournal.clerk.viewer/get-viewers of form (defn ensure-wrapped-with-viewers ([x] (ensure-wrapped-with-viewers (get-viewers (get-*ns*)) x)) ([viewers x] (-> x ensure-wrapped (update :nextjournal/viewers (fn [x-viewers] (or x-viewers viewers))))))
Missing hash for dep clojure.core/class of form (defrecord ViewerFn [form])
Missing hash for dep nextjournal.clerk.viewer/add-viewers of form (defn apply-viewers* [wrapped-value] (when (empty? (->viewers wrapped-value)) (throw (ex-info "cannot apply empty viewers" {:wrapped-value wrapped-value}))) (let [viewers (->viewers wrapped-value) {:as viewer, viewers-to-add :add-viewers, :keys [render-fn transform-fn]} (viewer-for viewers wrapped-value) transformed-value (cond-> (ensure-wrapped-with-viewers viewers (cond-> (-> wrapped-value (dissoc :nextjournal/viewer) (assoc :nextjournal/applied-viewer viewer)) transform-fn transform-fn)) viewers-to-add (update :nextjournal/viewers add-viewers viewers-to-add)) wrapped-value' (cond-> transformed-value (-> transformed-value ->value wrapped-value?) (merge (->value transformed-value)))] (if (and transform-fn (not render-fn)) (recur wrapped-value') (-> wrapped-value' (assoc :nextjournal/viewer viewer) (merge (->opts wrapped-value))))))
Missing hash for dep nextjournal.clerk.viewer/present+paginate-children of form (defn present* [{:as wrapped-value, :keys [path !budget store!-wrapped-value], :nextjournal/keys [viewers]}] (when (empty? viewers) (throw (ex-info "cannot present* with empty viewers" {:wrapped-value wrapped-value}))) (when store!-wrapped-value (store!-wrapped-value wrapped-value)) (let [{:as wrapped-value-applied, :nextjournal/keys [presented?]} (apply-viewers* wrapped-value) xs (->value wrapped-value-applied)] (when (and !budget (not presented?)) (swap! !budget (fn* [%1] (max (dec %1) 0)))) (-> (merge (->opts wrapped-value-applied) (when (empty? path) (select-keys wrapped-value [:present-elision-fn])) (with-viewer (->viewer wrapped-value-applied) (cond presented? wrapped-value-applied (string? xs) (present+paginate-string wrapped-value-applied) (and xs (seqable? xs)) (present+paginate-children wrapped-value-applied) :else xs))) process-wrapped-value)))
Missing hash for dep nextjournal.clerk.viewer/transform-result of form (def result-viewer {:name (quote nextjournal.clerk.viewer/result-viewer), :render-fn (quote nextjournal.clerk.render/render-result), :transform-fn transform-result})
```
which I don't get on this branch.